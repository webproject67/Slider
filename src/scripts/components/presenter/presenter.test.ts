import Presenter from './presenter';

const main = document.createElement('div');
main.id = 'banana';

const state = {
  flag: true,
  from: -10000,
  fromPercent: 0,
  max: 100,
  min: 0,
  progress: true,
  range: 'one',
  draft: 0,
  start: 1,
  configuring: 0,
  scale: true,
  step: 1,
  to: -10000,
  toPercent: 100,
  view: 'horizontal',
};

describe('snapshot', () => {
  test('renders correctly element1', () => {
    expect(new Presenter(main, state).model).toMatchSnapshot();
  });

  test('renders correctly element2', () => {
    state.start = 0;
    state.min = 10;
    state.max = 50;
    state.step = 10;
    state.flag = false;
    state.progress = false;
    expect(new Presenter(main, state).model).toMatchSnapshot();
  });

  test('renders correctly element3', () => {
    state.configuring = 1;
    state.scale = false;
    expect(new Presenter(main, state).model).toMatchSnapshot();
  });
});
