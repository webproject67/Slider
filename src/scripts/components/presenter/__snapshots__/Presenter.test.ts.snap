// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot renders correctly element1 1`] = `
Model {
  "observers": Array [
    [Function],
  ],
  "state": Object {
    "configuring": 0,
    "draft": 0,
    "flag": true,
    "from": -10000,
    "fromPercent": 0,
    "max": 100,
    "min": 0,
    "progress": true,
    "range": "one",
    "scale": true,
    "start": 0,
    "step": 1,
    "to": -10000,
    "toPercent": 100,
    "view": "horizontal",
  },
}
`;

exports[`snapshot renders correctly element2 1`] = `
Model {
  "observers": Array [
    [Function],
  ],
  "state": Object {
    "configuring": 0,
    "draft": 0,
    "flag": false,
    "from": -10000,
    "fromPercent": 0,
    "max": 50,
    "min": 10,
    "progress": false,
    "range": "one",
    "scale": true,
    "start": 0,
    "step": 10,
    "to": -10000,
    "toPercent": 100,
    "view": "horizontal",
  },
}
`;

exports[`snapshot renders correctly element3 1`] = `
Model {
  "observers": Array [
    [Function],
  ],
  "state": Object {
    "configuring": 1,
    "draft": 0,
    "flag": false,
    "from": -10000,
    "fromPercent": 0,
    "max": 50,
    "min": 10,
    "progress": false,
    "range": "one",
    "scale": false,
    "start": 0,
    "step": 10,
    "to": -10000,
    "toPercent": 100,
    "view": "horizontal",
  },
}
`;
